### PR TITLE
qa: switch to use merge fragment to filter the kclient

### PR DESCRIPTION
**Pull request by lxbsz**
PR https://github.com/ceph/ceph/pull/48183 have add the merge fragment support. This will switch to use fragment to simplify the code.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by lxbsz** [comment]
All the relevant [qa tests](https://pulpito.ceph.com/xiubli-2023-03-29_05:58:17-fs:upgrade:mds_upgrade_sequence-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/) are running.

**Comment by lxbsz** [comment]
Only updated the commit comments.

**Review by dparmar18 — APPROVED** [review]
LGTM

**Comment by dparmar18** [comment]
> All the relevant [qa tests](https://pulpito.ceph.com/xiubli-2023-03-29_05:58:17-fs:upgrade:mds_upgrade_sequence-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/) are running.

all dead due to `'7548196cdacea28b374443a9304dc4f8702e673e' not found in repo: https://github.com/lxbsz/ceph.git!`

**Comment by batrick** [comment]
@lxbsz also have a look at https://github.com/ceph/ceph/pull/46882

I hope to revive that PR in the near future.

**Review by batrick — APPROVED** [review]

**Comment by lxbsz** [comment]
> @lxbsz also have a look at #46882
> 
> I hope to revive that PR in the near future.

@batrick Yeah, sure I will review your PR today and will revive this one once after I successfully run the tests later.

Thanks,

**Comment by rishabh-d-dave** [comment]
@lxbsz Should this be put through QA?

**Comment by lxbsz** [comment]
> @lxbsz Should this be put through QA?

@rishabh-d-dave  This is not ready yet. Thanks.

**Comment by dparmar18** [comment]
Hi @lxbsz, this might've failed in QA run since you've added the `.qa/cephfs/mount/` symlink directly inside `qa/suites/fs/upgrade/mds_upgrade_sequence` but it needs to be inside `qa/suites/fs/upgrade/mds_upgrade_sequence/tasks` since the scope of function `is_fuse()` is limited to local dir and therefore `qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml` won't be able to find it.

Looking at the dir structure; you'd also need to make name changes inside `tasks/` as well, it would be now `0-from`, `1-mount.yaml`, `2-volume` and so on. Also reading this [comment](https://github.com/ceph/ceph/commit/898316a72c6fd5729e6b574fade6ba8adc3ca25a#diff-63be9c532a24d81b6391bb01e3d994eef8eb6a948154c227164063e6acd592f7R3-R6) from https://github.com/ceph/ceph/commit/898316a72c6fd5729e6b574fade6ba8adc3ca25a makes me think if that is needed here too. Does the distro kernels now have the newops fixes?

**Comment by lxbsz** [comment]
> Hi @lxbsz, this might've failed in QA run since you've added the `.qa/cephfs/mount/` symlink directly inside `qa/suites/fs/upgrade/mds_upgrade_sequence` but it needs to be inside `qa/suites/fs/upgrade/mds_upgrade_sequence/tasks` since the scope of function `is_fuse()` is limited to local dir and therefore `qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml` won't be able to find it.
> 
> Looking at the dir structure; you'd also need to make name changes inside `tasks/` as well, it would be now `0-from`, `1-mount.yaml`, `2-volume` and so on. Also reading this [comment](https://github.com/ceph/ceph/commit/898316a72c6fd5729e6b574fade6ba8adc3ca25a#diff-63be9c532a24d81b6391bb01e3d994eef8eb6a948154c227164063e6acd592f7R3-R6) from [898316a](https://github.com/ceph/ceph/commit/898316a72c6fd5729e6b574fade6ba8adc3ca25a) makes me think if that is needed here too. Does the distro kernels now have the newops fixes?

Sounds correct. I have fixed it. @dparmar18 Please take a look. Thanks.

**Review by dparmar18 — APPROVED** [review]

**Comment by dparmar18** [comment]
jenkins test make check

File: qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-tasks:
-- kclient:
-- print: "**** done client"
```
Comment (batrick) [inline qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml:3]: So I'm looking at this change again. Does this PR produce any functional difference? Is the goal to start testing all kernel branches (stock/testing) we normally do?

Please note that premerge/postmerge scripts are not really free. Structuring the matrix this way can blow up the matrix with jobs that will always be pruned.

File: qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/1-mount [hunk 2]
```diff
@@ -0,0 +1 @@
+.qa/cephfs/mount/
```
Comment (batrick) [inline qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/1-mount:1]: But I'm generally happy to see our suites consistently use a link to common yamls.

Comment (batrick) [reply qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/1-mount:1]: This moves the mount task to before the file system is created by `1-volume`. I don't think that's intentional?

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml (hunk 1 above)
Comment (lxbsz) [reply qa/suites/fs/upgrade/mds_upgrade_sequence/tasks/2-client.yaml:3]: I think we don't need this PR any more. The code changed a lot, but not sure why it didn't generated any conflicts in this PR.

The new code has added the `ceph-fuse` client.

**Pull request closed without merging**